base/regex.jl

```diff
@@ -363,13 +363,29 @@ function endswith(s::SubString{String}, r::Regex)
     return PCRE.exec_r(r.regex, s, 0, r.match_options | PCRE.ENDANCHORED)
 end
 
+
+_isASCIIletter(c) = ('A' <= c) & (bitrotate(reinterpret(Int32, c) & xor(32 << (32-8), -1), -32+8) <= Int32('Z'))
```

Comment:
Involving `bitrotate` to check for an ASCII char doesn't look good to me.

I also share the concern about combining `startswith` and `chopprefix` that was already pointed out.

I would ACK and merge the docs changes, but I am not at all convinced about the "optimizations". Maybe scale this PR back to the doc improvements (which then can be merged quickly), and then we can argue about the functionality changes in a separate PR.